src/librustc_mir/transform/qualify_min_const_fn.rs

```diff
@@ -324,10 +326,17 @@ fn check_terminator(
             check_operand(tcx, value, span, def_id, body)
         },
 
-        TerminatorKind::FalseEdges { .. } | TerminatorKind::SwitchInt { .. } => Err((
+        | TerminatorKind::FalseEdges { .. }
+        | TerminatorKind::SwitchInt { .. }
+        if !tcx.features().const_if_match
+        => Err((
             span,
             "loops and conditional expressions are not stable in const fn".into(),
         )),
+        | TerminatorKind::FalseEdges { .. }
+        | TerminatorKind::SwitchInt { .. }
+        => Ok(()),
+
         | TerminatorKind::Abort | TerminatorKind::Unreachable => {
```

Comment:
@oli-obk Also if you remember, what was the reason for disallowing `::Abort | ::Unreachable`?

Reply:
probably because we have no const fn code actually hitting it and thus there being no way to test it

Reply:
Can't we use `-Z panic=abort`?

Reply:
that just removes the landing pads, but still goes through the regular panic invocation I think. It just doesn't unwind in the end.

Reply:
Abort terminators are generated on `#[unwind(aborts)]` functions when compiling with `-C panic=unwind`, which I believe can happen for `const fn`.

Reply:
I opened https://github.com/rust-lang/rust/issues/66557